File: Python/Armstrong.py
i=0
result=0
n = int(input("please give a number : "))
number1 = n
temp = n
while n!=0:
    n = (n//10)
    i=i+1;
while number1!=0:
    n=number1%10
    result=result+pow(n,i)
    number1=number1//10
if temp==result:
    print("number is armstrong")
else:
    print("number is not armstrong") 

File: C/Armstrong.c
#include <stdio.h>
#include <math.h>

// Function to calculate the number of digits in a number
int countDigits(int num) {
    int count = 0;
    while (num != 0) {
        num /= 10;
        count++;
    }
    return count;
}

// Function to check if a number is an Armstrong number
int isArmstrong(int num) {
    int originalNum = num;
    int digitCount = countDigits(num);
    int sum = 0;

    while (num > 0) {
        int digit = num % 10;
        sum += pow(digit, digitCount);
        num /= 10;
    }

    return (sum == originalNum);
}

int main() {
    int number;

    printf("Enter a number: ");
    scanf("%d", &number);

    if (isArmstrong(number)) {
        printf("%d is an Armstrong number.\n", number);
    } else {
        printf("%d is not an Armstrong number.\n", number);
    }

    return 0;
}
